crates/turborepo-lib/src/run/watch.rs

```diff
@@ -337,7 +337,9 @@ impl WatchClient {
                 {
                     // Shut down the tasks for the run
                     stopper.stop().await;
-                    run_task.abort();
+                    // Run should exit shortly after we stop all child tasks, wait for it to finish
+                    // to ensure all messages are flushed.
+                    let _ = run_task.await;
```

Comment:
We don't `abort` as that seems to still gives us messages in the pipes. I think this is due to the TUI channels being from `std::sync::mpsc` instead of `tokio::sync::mpsc` (For [now](https://github.com/vercel/turborepo/pull/9132)).